### PR TITLE
feat: updated XCM logic to V3 for Shiden, Statemine and Kusama

### PR DESCRIPTION
**Pull request by impelcrypto**
**Pull Request Summary**

* feat: updated XCM logic to V3
  * Kusama
  * Statemine
  * Shiden 
  * Phala
  * Bitfrost
  * test results: https://docs.google.com/spreadsheets/d/1jmTF4VUVSeELAAMGdOrx72Rmc27jlVPxdGpdvCSXy2E/edit#gid=0
* fix: fixed minor styling issues
 
**Check list**

- [ ] contains breaking changes
- [ ] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**Release notes:**

* feat: updated XCM logic to V3 for Shiden and Kusama

**This pull request makes the following changes:**

**Changes**

* feat: updated XCM logic to V3
* fix: fixed minor styling issues

### CODE REVIEW

**Review by niklabh — APPROVED** [review]

**Comment by Kahonnohak** [comment]
Worked with KAR on Portal but it doesn't seem to be working on Nova, is this a separate issue?

**Comment by impelcrypto** [comment]
@Kahonnohak I think it should be fine as long as deposit/withdrawal works on substrate wallets like Polkadot.js

**Comment by Kahonnohak** [comment]
Now KSM works, lets merge and do more tests on staging

**Review by Kahonnohak — APPROVED** [review]
LGTM

File: src/v2/repositories/implementations/XcmRepository.ts [hunk 1]
```diff
@@ -215,6 +240,17 @@ export class XcmRepository implements IXcmRepository {
       ],
     };
 
+    console.log('XcmRepository');
```
Comment (bobo-k2) [inline src/v2/repositories/implementations/XcmRepository.ts:243]: Is this still needed or it was just debug log?

File: src/v2/repositories/implementations/XcmRepository.ts [hunk 2]
```diff
@@ -282,8 +318,11 @@ export class XcmRepository implements IXcmRepository {
     const config = await api.query.xcAssetConfig.assetIdToLocation<Option<AssetConfig>>(token.id);
 
     // return config.unwrap().v1;
+    const { isV3 } = this.getXcmVersion(source);
     const formattedAssetConfig = JSON.parse(config.toString());
-    return formattedAssetConfig.v1;
+
+    console.log('formattedAssetConfig', formattedAssetConfig);
```
Comment (bobo-k2) [inline src/v2/repositories/implementations/XcmRepository.ts:324]: Same as the above

**Review by bobo-k2 — APPROVED** [review]
LGTM

File: src/v2/repositories/implementations/XcmRepository.ts (hunk 1 above)
Comment (impelcrypto) [reply src/v2/repositories/implementations/XcmRepository.ts]: Thanks, I've fixed in [src/v2/repositories/implementations/XcmRepository.ts](https://github.com/AstarNetwork/astar-apps/pull/758/files/ecd8fd448d629222924992502e8b06c6817d23da#diff-4cba2fed53f9b329ee1c759339d4d0d7e7d3a56f39081ee13513286accab7f2b) commit.

**Pull request merged**